src/state/queries/feed.ts

```diff
@@ -284,14 +309,34 @@ export function usePinnedFeedsInfos() {
           }),
       )
 
-      // The returned result will have the original order.
-      const result = [hasSession ? FOLLOWING_FEED_STUB : DISCOVER_FEED_STUB]
+      const result = [hasSession ? HOME_FEED_STUB : PWI_DISCOVER_FEED_STUB]
+
       await Promise.allSettled([feedsPromise, ...listsPromises])
-      for (let pinnedUri of pinnedUris) {
-        if (resolved.has(pinnedUri)) {
-          result.push(resolved.get(pinnedUri))
+
+      // if primary algo is enabled and was fetched, add it to the front of the list
+      if (primaryAlgo?.enabled && primaryAlgo?.uri) {
+        const feedInfo = resolved.get(primaryAlgo.uri)
+        if (feedInfo) {
+          feedInfo.isPrimaryAlgorithm = true
+          result.unshift(feedInfo)
         }
       }
+
+      const pinnedUrisSansPrimary = pinnedUris.filter(uri => {
+        if (primaryAlgo?.enabled) {
+          return uri !== primaryAlgo?.uri
+        }
+        return true
+      })
```

Comment:
This is defensive, as are a couple other spots. Example: 3p app allows user to pin their primary algo.